### DISCUSSION TITLE
amqp Message JmsListener

### DISCUSSION THREAD

**Discussion by chenyiy** [rabbitmq/rabbitmq-jms-client#145] [General]
``` java
    @JmsListener(destination = "treeQueue")
    public JmsResponse sessssnd(Message message) throws Exception {
        RMQDestination destination = (RMQDestination) message.getJMSReplyTo();
        destination.setDeclared(true);
        String response = "response";
        return JmsResponse.forDestination(response, destination);
    }
```

```java
        RMQDestination jmsDestination = new RMQDestination();
        jmsDestination.setAmqp(true);
        jmsDestination.setAmqpExchangeName(exchange);
        jmsDestination.setAmqpRoutingKey(routingKey);
        TextMessage msg = session.createTextMessage("this test message");
        msg.setJMSType("TextMessage");
        msg.setStringProperty("JMSType", "TextMessage");
        jmsTemplate.convertAndSend(jmsDestination,msg);
```

2021-11-22 19:57:28.944  WARN 25156 --- [tContainer#1-69] o.s.j.l.DefaultMessageListenerContainer  : Setup of JMS message listener invoker failed for destination 'treeQueue' - trying to recover. Cause: null

**Comment by acogoluegnes** [comment]
We need more information. Could you please share a standalone project that reproduces the problem?

Thanks.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(
